onnxscript/version_converter/version_converter.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+from __future__ import annotations
+
+from onnxscript import ir
+
+### Adapters
+
+# Compatibility Adapter
+def adapter_compatible(op: ir.Node, target_opset):
+    op.version = target_opset
```

Comment:
Setting the version on a node need not be part of the adapter logic. It should be in the converter logic that calls the adapter, since this logic is the same for all adapters. No point in duplicating this line in every adapter. That would also make a compatible adapter an identity function, and we don't even need to register one for compatible extensions,